#10036 Replace codename Merge with Bellatrix (1st step)

leolara opened the pull request:
<!-- Thanks for sending a PR! Before submitting:

1. If this is your first PR, check out our contribution guide here https://docs.prylabs.network/docs/contribute/contribution-guidelines
   You will then need to sign our Contributor License Agreement (CLA), which will show up as a comment from a bot in this pull request after you open it. We cannot review code without a signed CLA.
2. Please file an associated tracking issue if this pull request is non-trivial and requires context for our team to understand. All features and most bug fixes should have
   an associated issue with a design discussed and decided upon. Small bug
   fixes and documentation improvements don't need issues.
3. New features and bug fixes must have tests. Documentation may need to
   be updated. If you're unsure what to update, send the PR, and we'll discuss
   in review.
4. Note that PRs updating dependencies and new Go versions are not accepted.
   Please file an issue instead.
-->

**What type of PR is this?**

Refactoring

**What does this PR do? Why is it needed?**

Replace codename Merge with Bellatrix 

**Which issues(s) does this PR fix?**

Fixes #10036 

**Other notes for review**

It does not complete 100% the task #10036 another PR will be necessary

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prysmaticlabs/prysm?pullRequest=10044) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prysmaticlabs/prysm?pullRequest=10044) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prysmaticlabs/prysm?pullRequest=10044) it.</sub>

leolara commented:
It is not passing in my local system the test `TestEndToEnd_MinimalConfig_ValidatorAtCurrentRelease/chain_started` but it does not pass it also without any changes, so I guess I have a local problem, and this is not breaking any tests.

The tests I am running is `bazel test //testing/endtoend:go_default_test` I am not sure if I need to run other tests.

rkapka commented on beacon-chain/state/stateutil/state_hasher.go:
```diff
@@ -371,15 +371,15 @@ func (h *stateRootHasher) ComputeFieldRootsWithHasherAltair(ctx context.Context,
 
 // ComputeFieldRootsWithHasherMerge hashes the provided merge state and returns its respective field roots.
-func (h *stateRootHasher) ComputeFieldRootsWithHasherMerge(ctx context.Context, state *ethpb.BeaconStateMerge) ([][]byte, error) {
+func (h *stateRootHasher) ComputeFieldRootsWithHasherMerge(ctx context.Context, state *ethpb.BeaconStateBellatrix) ([][]byte, error) {
```
```suggestion
func (h *stateRootHasher) ComputeFieldRootsWithHasherBellatrix(ctx context.Context, state *ethpb.BeaconStateBellatrix) ([][]byte, error) {
```

rkapka commented on beacon-chain/state/stateutil/state_hasher.go:
```diff
@@ -371,15 +371,15 @@ func (h *stateRootHasher) ComputeFieldRootsWithHasherAltair(ctx context.Context,
 }
 
 // ComputeFieldRootsWithHasherMerge hashes the provided merge state and returns its respective field roots.
```
```suggestion
// ComputeFieldRootsWithHasherBellatrix hashes the provided merge state and returns its respective field roots.
```

rkapka commented:
> Following files need to be properly formatted:
config/params/mainnet_config.go
proto/prysm/v1alpha1/beacon_state.pb.go

rkapka commented:
Hey @leolara , you still need to run `gofmt` on a few files:

> Following files need to be properly formatted:
config/params/mainnet_config.go
proto/prysm/v1alpha1/beacon_state.pb.go
runtime/version/fork.go

leolara commented:
@rkapka thank you for reminding me that. This PR is marked as WIP. The merge of develop into has caused me so git problems with a commit I did at the same time

leolara commented:
There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.

Is there a bazel way of running gofmt? Or I just run it normally?

potuz commented:
> There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.
> 
> Is there a bazel way of running gofmt? Or I just run it normally?

You can run it normally or even as a pre commit hook. I recommend you mark the PR as draft, since then reviewers will get an automated email when you mark it requesting a review

leolara commented:
@Potuz

> > There are lots of things to rename, I will let you know when I think I am finished, so we can review @rkapka . I will try to also rebase on develop after that so it is fresher.
> > Is there a bazel way of running gofmt? Or I just run it normally?
> 
> You can run it normally or even as a pre commit hook. I recommend you mark the PR as draft, since then reviewers will get an automated email when you mark it requesting a review

Thanks, I changed it to draft

leolara commented:
I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣

terencechain commented:
> I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣

If it makes it easier, you don't have to replace everything at once. Open a PR package by package is perfectly fine

terencechain commented on beacon-chain/core/blocks/payload_test.go:
```diff
@@ -12,7 +12,7 @@ import (
 )
 
-func Test_MergeComplete(t *testing.T) {
+func Test_BellatrixComplete(t *testing.T) {
```
This refers to the action of merge

leolara replied:
Thanks @terencechain , this one slipped :-)

leolara commented:
> > I want to note that there is more than a thousand "merge" string instances in the code, and that some times it actually means the action of merge, as opposed to the hard-fork. So I have to read it 🤣
> 
> If it makes it easier, you don't have to replace everything at once. Open a PR package by package is perfectly fine

I am doing it more subject by subject, like I am now changing comments and will send that as a commit. If you think the PR is going to get too long to review, then I could do it in several PRs. I think I am kind of half way

leolara commented:
In order to do not make the review too tedious and complicated, I have decided to stop considering it WIP, get it reviewed and merged, and then I will continue with a follow up PR with the rest of the renaming.

I have rebased on develop, no need to merge develop now.

Please review @terencechain or other :-)

terencechain commented on config/params/mainnet_config.go:
```diff
@@ -237,6 +237,6 @@ var mainnetBeaconConfig = &BeaconChainConfig{
 	MinSyncCommitteeParticipants: 1,
 
-	// Merge
+	// Bellatrix
```
This is a close one, but I think I like `Merge` better here. `TerminalBlockHashActivationEpoch` is used for merging the two chains

leolara replied:
Done

prestonvanloon replied:
@leolara it still says Bellatrix. Did you intend to change it back to "Merge" per Terence's comment?

leolara replied:
Done

terencechain commented on proto/prysm/v1alpha1/beacon_block.proto:
```diff
@@ -33,7 +33,7 @@ message GenericSignedBeaconBlock {
 
-        // Representing a signed, post-Merge fork beacon block.
+        // Representing a signed, post-Bellatrix fork beacon block.
         SignedBeaconBlockMerge merge = 3;
```
```suggestion
        SignedBeaconBlockBellatrix bellatrix = 3;
```

Just a reminder here so we don't forget. We can do this in a follow-up PR

leolara commented:
@terencechain I rebased to fix conflict, it seems it needs further action from mantainers to merge

prestonvanloon commented on beacon-chain/db/kv/schema.go:
```diff
@@ -46,8 +46,8 @@ var (
 	// Objects that are only compatible with specific forks should be prefixed with such keys.
-	altairKey = []byte("altair")
-	mergeKey  = []byte("merge")
+	altairKey    = []byte("altair")
+	bellatrixKey = []byte("bellatrix")
```
This is going to break everyone on testnets. Do not rename database keys without a migration.

leolara replied:
Reverted

prestonvanloon commented on proto/prysm/v1alpha1/beacon_state.pb.go:
```diff
@@ -1073,7 +1073,7 @@ func (x *SyncAggregatorSelectionData) GetSubcommitteeIndex() uint64 {
 }
 
-type BeaconStateMerge struct {
+type BeaconStateBellatrix struct {
```
Renaming in the generated code is not ideal. Please re-run `./hack/update-go-pbs.sh` after modifying proto files.

The reason is that `file_proto_prysm_v1alpha1_beacon_state_proto_rawDesc` attribute should be updated as well and it's some hex bytes so rebuilding the pb.go files is really the only way to do it. You may need to run gofmt or goimports to sort the imports for the linter.

leolara replied:
ok

leolara replied:
Done, many files regenerated and pushed

leolara commented:
@prestonvanloon @terencechain @rkapka I think all comments done, rebased to resolve conflicts.

Do you want me to squash when merging or merge normally?

Please, I think we need to move this, as this touches so many files many PRs will conflict this one

leolara commented:
It seems the problems are not only conflicts with new additions, but also the use of symbols I have renamed in other PRs.

For example, I renamed version.Merge to version.Bellatrix, if someone uses version.Merge and merges a PR, then that symbol does not exist in my PR.

I think we should freeze merging other PRs until this is merged or it will be very difficult

prestonvanloon commented:
@leolara You need to run gofmt and goimports after regenerating protofiles.

```
gofmt -w **/*.go
goimports -w **/*.go
```

Git/Github is not letting me push to your branch. Please fix and we can merge quickly.

leolara commented:
@prestonvanloon I ran those commands, but there were no effect, it seems my branch is clean or the problem is other

leolara commented:
@prestonvanloon I changed a setting try to push to my branch now
